src/Utilities/VulnerabilityRoutes.test.js

```diff
@@ -23,98 +22,64 @@ jest.mock('@redhat-cloud-services/frontend-components/AsyncComponent', () => ({
-                getSystems.mockReturnValueOnce(Promise.resolve({ meta: { total_items: 0 } }));
-                render(<ComponentWithContext>
-                    <InsightsElement element={TestElement} />
-                </ComponentWithContext>);
-    
-                await waitFor(() => {
-                    expect(
-                        screen.getByLabelText('Zero state')
-                    ).toBeVisible();
-                });
+            it('Should show zero state when there is no conventional, regardless of edge system', async () => {
+                let isEdgeParityEnabled = false;
+                let hasEdgeDevices = true;
+                let hasConventionalSystems = false;
+                expect(checkForAccountSystems(isEdgeParityEnabled, hasEdgeDevices, hasConventionalSystems)).toBe(false)
+
             });
-            it('Should element when there is a conventional system, regardless of edge system', async () =>{
-                render(<ComponentWithContext>
-                    <InsightsElement element={TestElement} />
-                </ComponentWithContext>);
-    
-                await waitFor(() => {
-                    expect(
-                        screen.getByLabelText('Insights element')
-                    ).toBeVisible();
-                });
+            
+            it('Should element when there is a conventional system, regardless of edge system', async () => {
+                let isEdgeParityEnabled = false;
+                let hasEdgeDevices = true;
+                let hasConventionalSystems = true;
+                expect(checkForAccountSystems(isEdgeParityEnabled, hasEdgeDevices, hasConventionalSystems)).toBe(true)
             });
-        });
+        })
 
         describe(('edge parit enabled'), () => {
-            beforeEach(() => {
-                useFeatureFlag.mockReturnValue(true);
-            });
-            it('Should show zero state when there is no conventional and edge systems', async () =>{
-                getSystems.mockReturnValueOnce(Promise.resolve({ meta: { total_items: 0 } }));
-                checkEdgePresence.mockReturnValueOnce(Promise.resolve(false));
-                render(<ComponentWithContext>
-                    <InsightsElement element={TestElement} />
-                </ComponentWithContext>);
-    
-                await waitFor(() => {
-                    expect(
-                        screen.getByLabelText('Zero state')
-                    ).toBeVisible();
-                });
+            it('Should show zero state when there is no conventional and no edge systems', async () => {
+                let isEdgeParityEnabled = true;
+                let hasEdgeDevices = false;
+                let hasConventionalSystems = false;
+                expect(checkForAccountSystems(isEdgeParityEnabled, hasEdgeDevices, hasConventionalSystems)).toBe(false)
```

Comment:
As we have migrated to RTL, would it be better to assert on the UI element itself instead of the checkForAccountSystems function result?

Reply:
Originally that was the plan, however the routing page is essentially just a fed module now. So we're really just making sure the right state renders when passing in the correct items